### PR TITLE
Added RealmObjectSchema.getFieldType

### LINKED ISSUE #1883: Checking type with DynamicRealm's RealmSchema
Hello, 

Previously I'm using Table to check the type of the column now I'm trying to migrate to DynamicRealm because [direct access to Table will be deprecated soon.](https://github.com/realm/realm-java/issues/1876)

I think I can get the name of the field with using 

```
RealmObjectSchema realmObjectSchema = dynamicRealm.getSchema().get(selectedTableName);
Set<String> fieldNames = realmObjectSchema.getFieldNames();
```

But how to get the field types?

Thank you!

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #1883 

@realm/java

### CODE REVIEW

**Comment by beeender** [comment]
:+1:

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java [hunk 1]
```diff
@@ -579,6 +579,22 @@ public void testGetFieldNames() {
...
+        assertEquals(RealmFieldType.DATE, schema.getFieldType(AllJavaTypes.FIELD_DATE));
+        assertEquals(RealmFieldType.DOUBLE, schema.getFieldType(AllJavaTypes.FIELD_DOUBLE));
+        assertEquals(RealmFieldType.FLOAT, schema.getFieldType(AllJavaTypes.FIELD_FLOAT));
+        assertEquals(RealmFieldType.OBJECT, schema.getFieldType(AllJavaTypes.FIELD_OBJECT));
+        assertEquals(RealmFieldType.LIST, schema.getFieldType(AllJavaTypes.FIELD_LIST));
+        assertEquals(RealmFieldType.INTEGER, schema.getFieldType(AllJavaTypes.FIELD_BYTE));
+        assertEquals(RealmFieldType.INTEGER, schema.getFieldType(AllJavaTypes.FIELD_SHORT));
+        assertEquals(RealmFieldType.INTEGER, schema.getFieldType(AllJavaTypes.FIELD_INT));
+        assertEquals(RealmFieldType.INTEGER, schema.getFieldType(AllJavaTypes.FIELD_LONG));
+    }
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: What about a test where you look up a non-existing field?

Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: Good idea, i'll add it.

Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: Added

**Comment by kneth** [comment]
:+1:

**Pull request merged**